src/data/polynomial/coeff.lean

```diff
@@ -96,6 +96,15 @@ end
 @[simp] lemma mul_coeff_zero (p q : R[X]) : coeff (p * q) 0 = coeff p 0 * coeff q 0 :=
 by simp [coeff_mul]
 
+/-- `constant_coeff p` returns the constant term of the polynomial `p`,
+  defined as `coeff p 0`. This is a ring homomorphism. -/
+@[simps] def constant_coeff : R[X] →+* R :=
+{ to_fun := λ p, coeff p 0,
+  map_one' := by rw [coeff_one, if_pos rfl],
+  map_mul' := mul_coeff_zero,
+  map_zero' := coeff_zero 0,
+  map_add' :=  λ p q, coeff_add p q 0 }
+
```

Comment:
You don't mean coercion to function, do you? Because [eval is the simp normal form in that case](https://leanprover-community.github.io/mathlib_docs/data/polynomial/algebra_map.html#polynomial.coe_aeval_eq_eval).

Actually there's already `eval_ring_hom` that is already a ring_hom but with commutativity assumption, because it evaluate at any element, not just central ones. So I added
```
lemma eval_ring_hom_zero : eval_ring_hom 0 = constant_coeff :=
fun_like.ext _ _ $ λ p, p.coeff_zero_eq_eval_zero.symm
```
but I'm not sure if it should be a simp lemma.